chat-server/server.js

```diff
@@ -21,8 +23,68 @@ const welcomeMessage = {
 //We will start with one message in the array.
 const messages = [welcomeMessage];
 
-app.get("/", (request, response) => {
-  response.sendFile(__dirname + "/index.html");
+app.get("/", (req, res) => {
+  res.sendFile(__dirname + "/index.html");
+});
+
+app.post("/messages", (req, res) => {
+  const newMessage = req.body;
+  if (
+    !newMessage.hasOwnProperty("text") ||
+    !newMessage.hasOwnProperty("from") ||
+    newMessage.from === "" ||
+    newMessage.text === "" ||
+    typeof newMessage.text != "string" ||
+    typeof newMessage.from != "string"
+  ) {
```

Comment:
If you want to go one further step, you can create your validation fun and give single responsibility to it. It looks like this:
```
function validateMessage(newMessage) {
     //your validation returns boolean
 }

app.post("/messages", (req, res) => {
  const newMessage = req.body;

  const isMessageValid = validateMessage(newMessage

  if (!isMessageValid)) {
    res.status(400).send("wrong input");
    return;
  }
// Rest of your code...

Reply:
I have modified the route function. Could you have a look please?